azuracast/liquidsoap_custom.liq

```diff
@@ -1,488 +1,512 @@
     live_dj := dj
-    
-    ret = list.hd(get_process_lines("curl -s --request POST --url http://web/api/internal/1/djon --form dj-user="^string.quote(dj)^" --form api_auth="^string.quote("AZURA_KEY")^""), default="")
-    log("AzuraCast Live Connected Response: #{ret}")
+
+    _ = azuracast_api_call(
+        timeout_ms=5000,
+        "djon",
+        json.stringify({user = dj})
+    )
 end
 
-def live_disconnected() = 
-    dj = !live_dj
-    
-    log("DJ Source disconnected! Current live DJ: #{dj}")
-    
-    ret = list.hd(get_process_lines("curl -s --request POST --url http://web/api/internal/1/djoff --form dj-user="^string.quote(dj)^" --form api_auth="^string.quote("AZURA_KEY")^""), default="")
-    log("AzuraCast Live Disconnected Response: #{ret}")
-    
+def live_disconnected() =
+    _ = azuracast_api_call(
+        timeout_ms=5000,
+        "djoff",
+        json.stringify({user = !live_dj})
+    )
+
     live_enabled := false
-    last_authenticated_dj := ""
     live_dj := ""
-end 
-
-# A Pre-DJ source of radio that can be broadcast if needed
-radio_without_live = radio
-ignore(radio_without_live)
+end
 
 # Live Broadcasting
-live = audio_to_stereo(input.harbor("/", id = "lahmacun_radio_input_streamer", port = 8005, auth = dj_auth, icy = true, icy_metadata_charset = "UTF-8", metadata_charset = "UTF-8", on_connect = live_connected, on_disconnect = live_disconnected, buffer = 5., max = 10.))
-ignore(output.dummy(live, fallible=true))
+live = input.harbor("/", id = "input_streamer", port = 8005, auth = dj_auth, icy = true, icy_metadata_charset = "UTF-8", metadata_charset = "UTF-8", on_connect = live_connected, on_disconnect = live_disconnected, buffer = 12.00, max = 15.00)
+
+def insert_missing(m) =
+    if m == [] then
+        [("title", "Live Broadcast"), ("is_live", "true")]
+    else
+        [("is_live", "true")]
+    end
+end
+live = map_metadata(insert_missing, live)
+
+sources = [ 
+    ({!live_enabled}, live),    #Live DJ
+    ({1w and 10h-10h58m}, once(playlist_szedd_a_labad)),             # Monday
+    ({1w and 11h11m-12h9m}, once(playlist_transverszia)),             # orig requested time 11:11 am 12:12 pm
+    ({1w and 13h-13h58m}, once(playlist_a_who_say)),
+    ({1w and 14h-15h28m}, once(playlist_korunk)),    
+    ({1w and 16h-16h58m}, once(playlist_szmuti_csorba)),
+    ({1w and 17h-17h58m}, once(playlist_paikka)),
+    ({1w and 18h-18h58m}, once(playlist_udcsi)),
+    ({1w and 19h-19h28m}, once(playlist_soundscape_diaries_emcsivel)),
+    ({1w and 19h30m-20h58m}, once(playlist_kolegak)),
+    ({1w and 21h-21h58m}, once(playlist_urmajmok)),      
+    ({1w and 22h-23h58m}, once(playlist_infinite_scroll)),      
+    ({1w and 13h30m-23h30m}, playlist_off_air_ambient),                        #ambient: Who Say - Infinite
+    ({2w and 12h-13h58m}, once(playlist_lahmaloudclouds)),
+    ({2w and 15h30m-16h58m}, once(playlist_melyvagas)),
+    ({2w and 17h-18h58m}, once(playlist_radical_happiness)),
```

Comment:
That's a bad typo. Thanks for finding it!

That's why it's good that we found a solution for an elegant handling of daylight saving time adjustments. I hope it will work smoothly.